// @aws-sdk/client-dynamodb/dist-es/commands/DescribeImportCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeImportInputFilterSensitiveLog, DescribeImportOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_0DescribeImportCommand, serializeAws_json1_0DescribeImportCommand, } from "../protocols/Aws_json1_0";
var DescribeImportCommand = (function (_super) {
    __extends(DescribeImportCommand, _super);
    function DescribeImportCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeImportCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "DynamoDBClient";
        var commandName = "DescribeImportCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeImportInputFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeImportOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeImportCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_0DescribeImportCommand(input, context);
    };
    DescribeImportCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_0DescribeImportCommand(output, context);
    };
    return DescribeImportCommand;
}($Command));
export { DescribeImportCommand };
