// @aws-sdk/lib-dynamodb/dist-es/commands/PutCommand.js
import { __awaiter, __extends, __generator } from "tslib";
import { PutItemCommand as __PutItemCommand, } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClientCommand } from "../baseCommand/DynamoDBDocumentClientCommand";
var PutCommand = (function (_super) {
    __extends(PutCommand, _super);
    function PutCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        _this.inputKeyNodes = [
            { key: "Item" },
            {
                key: "Expected",
                children: {
                    children: [{ key: "Value" }, { key: "AttributeValueList" }],
                },
            },
            { key: "ExpressionAttributeValues" },
        ];
        _this.outputKeyNodes = [
            { key: "Attributes" },
            { key: "ItemCollectionMetrics", children: [{ key: "ItemCollectionKey" }] },
        ];
        _this.clientCommand = new __PutItemCommand(_this.input);
        _this.middlewareStack = _this.clientCommand.middlewareStack;
        return _this;
    }
    PutCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        var _this = this;
        this.addMarshallingMiddleware(configuration);
        var stack = clientStack.concat(this.middlewareStack);
        var handler = this.clientCommand.resolveMiddleware(stack, configuration, options);
        return function () { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            return [2, handler(this.clientCommand)];
        }); }); };
    };
    return PutCommand;
}(DynamoDBDocumentClientCommand));
export { PutCommand };
