// @aws-sdk/lib-dynamodb/dist-es/commands/TransactGetCommand.js
import { __awaiter, __extends, __generator } from "tslib";
import { TransactGetItemsCommand as __TransactGetItemsCommand, } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClientCommand } from "../baseCommand/DynamoDBDocumentClientCommand";
var TransactGetCommand = (function (_super) {
    __extends(TransactGetCommand, _super);
    function TransactGetCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        _this.inputKeyNodes = [{ key: "TransactItems", children: [{ key: "Get", children: [{ key: "Key" }] }] }];
        _this.outputKeyNodes = [{ key: "Responses", children: [{ key: "Item" }] }];
        _this.clientCommand = new __TransactGetItemsCommand(_this.input);
        _this.middlewareStack = _this.clientCommand.middlewareStack;
        return _this;
    }
    TransactGetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        var _this = this;
        this.addMarshallingMiddleware(configuration);
        var stack = clientStack.concat(this.middlewareStack);
        var handler = this.clientCommand.resolveMiddleware(stack, configuration, options);
        return function () { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            return [2, handler(this.clientCommand)];
        }); }); };
    };
    return TransactGetCommand;
}(DynamoDBDocumentClientCommand));
export { TransactGetCommand };
